### PR TITLE
Bluetooth: Controller: Event overhead values plus Extended Adv duration fix

### PR DESCRIPTION
**Pull request by cvinayak**
- Remaining duration is now in us instead of ticks to avoid overflow when ticks have higher resolution
- Account for the added random delay when updating remaining duration
- Re-worked JIT re-scheduling update a bit to keep the random delay information and remaining duration correct
- Add EVENT_OVERHEAD_START_US to ticks_anchor when enabling advertising, so the first advertising event is not late due to calculations for aux packets etc.

Signed-off-by: Troels Nilsson <trnn@demant.com>

- Tune the EVENT_OVERHEAD_START_US for typical usecases on
nRF52833 SoC using the central_gatt_write and
peripheral_gatt_write samples.

- Manually run the samples for over 30 mins without any
assertions and use the actual profiled value that would
otherwise be printed on assertion.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Comment by Tronil** [comment]
Looks good with regards to the changes from PR 50031 :+1:

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c [hunk 1]
```diff
@@ -472,10 +472,13 @@ uint32_t lll_preempt_calc(struct ull_hdr *ull, uint8_t ticker_id,
 		 *    duration.
 		 * 3. Increase the preempt to start ticks for future events.
 		 */
-		return 1;
+		LL_ASSERT_MSG(0, "%s: Actual EVENT_OVERHEAD_START_US = %u\n",
+			      __func__, HAL_TICKER_TICKS_TO_US(diff));
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c:476]:
```suggestion
		LL_ASSERT_MSG(false, "%s: Actual EVENT_OVERHEAD_START_US = %u\n",
			      __func__, HAL_TICKER_TICKS_TO_US(diff));
```

File: subsys/bluetooth/controller/ll_sw/ull_adv.c [hunk 2]
```diff
@@ -1993,11 +2012,10 @@ void ull_adv_done(struct node_rx_event_done *done)
...
 		rx_hdr->rx_ftr.param_adv_term.status = BT_HCI_ERR_LIMIT_REACHED;
-	} else if (adv->ticks_remain_duration &&
-		   (adv->ticks_remain_duration <=
-		    HAL_TICKER_US_TO_TICKS((uint64_t)adv->interval *
-			ADV_INT_UNIT_US))) {
-		adv->ticks_remain_duration = 0;
+	} else if (adv->remain_duration_us &&
+		   (adv->remain_duration_us <=
+		    ((uint64_t)adv->interval * ADV_INT_UNIT_US))) {
+		adv->remain_duration_us = 0;
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_adv.c:2018]:
```suggestion
		adv->remain_duration_us = 0U;
```

File: subsys/bluetooth/controller/ll_sw/ull_adv.c [hunk 3]
```diff
@@ -2288,21 +2307,25 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t ticks_drift,
...
-		if (adv->ticks_remain_duration) {
-			uint32_t ticks_interval =
-				HAL_TICKER_US_TO_TICKS((uint64_t)adv->interval *
-						       ADV_INT_UNIT_US);
-			uint32_t ticks_elapsed = ticks_interval * (lazy + 1) +
-						 ticks_drift;
-
-			if (adv->ticks_remain_duration > ticks_elapsed) {
-				adv->ticks_remain_duration -= ticks_elapsed;
+		if (adv->remain_duration_us && adv->event_counter > 0) {
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_adv.c:2310]:
```suggestion
		if (adv->remain_duration_us && adv->event_counter > 0U) {
```

File: subsys/bluetooth/controller/ll_sw/ull_adv.c [hunk 4]
```diff
@@ -2288,21 +2307,25 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t ticks_drift,
...
+#if defined(CONFIG_BT_CTLR_JIT_SCHEDULING)
+			/* ticks_drift is always 0 with JIT scheduling, populate manually */
+			ticks_drift = adv->delay_at_expire;
+#endif /* CONFIG_BT_CTLR_JIT_SCHEDULING */
+			uint32_t interval_us = (uint64_t)adv->interval * ADV_INT_UNIT_US;
+			uint32_t elapsed_us = interval_us * (lazy + 1) +
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_adv.c:2316]:
```suggestion
			uint32_t elapsed_us = interval_us * (lazy + 1U) +
```

File: subsys/bluetooth/controller/ll_sw/ull_adv.c [hunk 5]
```diff
@@ -2288,21 +2307,25 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t ticks_drift,
...
+			if (adv->remain_duration_us > elapsed_us + interval_us +
+						      HAL_TICKER_TICKS_TO_US(random_delay)) {
+				adv->remain_duration_us -= elapsed_us;
 			} else {
-				adv->ticks_remain_duration = ticks_interval;
+				adv->remain_duration_us = interval_us;
 			}
 		}
+
+		adv->event_counter += (lazy + 1);
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_adv.c:2328]:
```suggestion
		adv->event_counter += (lazy + 1U);
```

File: subsys/bluetooth/controller/ll_sw/ull_adv.c [hunk 6]
```diff
@@ -2492,8 +2515,7 @@ static void adv_max_events_duration_set(struct ll_adv_set *adv,
 {
 	adv->event_counter = 0;
 	adv->max_events = max_ext_adv_evts;
-	adv->ticks_remain_duration =
-		HAL_TICKER_US_TO_TICKS((uint64_t)duration * 10 * USEC_PER_MSEC);
+	adv->remain_duration_us = (uint32_t)duration * 10 * USEC_PER_MSEC;
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_adv.c:2518]:
```suggestion
	adv->remain_duration_us = (uint32_t)duration * 10U * USEC_PER_MSEC;
```

File: subsys/bluetooth/controller/ll_sw/ull_adv.c [hunk 7]
```diff
@@ -2934,7 +2956,7 @@ static void init_set(struct ll_adv_set *adv)
 	adv->lll.chan_map = BT_LE_ADV_CHAN_MAP_ALL;
 	adv->lll.filter_policy = BT_LE_ADV_FP_NO_FILTER;
 #if defined(CONFIG_BT_CTLR_JIT_SCHEDULING)
-	adv->delay_remain = ULL_ADV_RANDOM_DELAY;
+	adv->delay = 0;
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_adv.c:2959]:
```suggestion
	adv->delay = 0U;
```

**Review by Thalley — DISMISSED** [review]
SOme minor comments that can be omitted

File: subsys/bluetooth/controller/ll_sw/ull_adv.c (hunk 7 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_adv.c]: done

**Review by Thalley — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**